src/H5FDs3comms.c

```diff
@@ -1664,17 +1689,17 @@ H5FD_s3comms_aws_canonical_request(char *canonical_request_dest, int _cr_size, c
 
         HDassert(node->magic == S3COMMS_HRB_NODE_MAGIC);
 
-        ret = HDsnprintf(tmpstr, 256, "%s:%s\n", node->lowername, node->value);
-        if (ret < 0 || ret >= 256)
+        ret = HDsnprintf(tmpstr, 1024, "%s:%s\n", node->lowername, node->value);
```

Comment:
This should really be "1024 + 1" since that's how big the buffer is, but I think the "1024 + 1" above should just be "1024" anyway and should be put behind some #define. Not really necessary to getting this PR merged, but it would be nice cleanup.

Reply:
The last element of the tmpstr (position: 1024) is the \0 character to ensure that tmpstr is properly NULL-terminated. This necessary for the function `HDstrlen(tmpstr)`.

Reply:
Sure, but snprintf should always NUL-terminate the buffer.

Reply:
Ah, you are right. It turns out that I misunderstood the man page of `snprintf()`. I always thought that the specified maximum size `size` is the maximum number of characters that will be set based on the format string and specified input strings while it is actually `size - 1` and the last position is always reserved for the NULL-termination. Anyway, this is very useful to know.

Reply:
This has been fixed.